GitHub_Learning/Feature/Onboarding/Presentation/OnboardingPresenter.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  OnboardingPresenter.swift
+//  GitHub_Learning
+//
+//  Created by Amin Ashhab on 09/11/2020.
+//  Copyright © 2020 Amin_Second_Test_Project. All rights reserved.
+//
+
+import Foundation
+import CleanCore
+import CleanPlatform
+
+protocol OnboardingPresenter: Presenter {
+    func updatePageControlFor(_ page: Int)
+    func updatePageTo(_ page: Int)
+    func updateButtonTitleFor(_ page: Int)
```

Comment:
Stejne jako u view procotolu, i tento bude potreba trochu upravit. View by ti melo davat vedet, kdyz uzivatel klidne na tlacitko / page control nebo swipne na novou page. To by mel reflektovat ten protocol:
```suggestion
    func next()
    func previous()
    func presentingPage(at index: Int)
```
Ty si budes interne drzet prave zobrazeny index a aktualizace button title bude v ramci toho.

Reply:
Asi nerozumím tomu previous, mám jen jeden button a to next, a proč se to má jmenovat presentingPage a né presentPage. volalo by se to teda: presenter.presentPage(at:).

Reply:
aha tak presentingPage už asi chápu, ale func previous má být pro previos button? Ten tam totiž mít nebudu.

Reply:
Ale mas tam page control, ktery ten previous button ma.